dapp/src/hooks/useCurrencySwapper.js

```diff
@@ -122,20 +133,35 @@ const useCurrencySwapper = ({
     callObject,
     swapAmount,
     minSwapAmount,
-    options = {}
+    txParams = {}
   ) => {
     return await callObject.mint(
       coinContract.address,
       swapAmount,
       minSwapAmount,
-      options
+      txParams
     )
   }
 
+  // Increases the given gas limit by the specified buffer
+  const increaseGasLimitByBuffer = (gasLimit, bufferToIncrease) => {
+    return Math.round(gasLimit * (1 + bufferToIncrease))
+  }
+
   const mintVaultGasEstimate = async (swapAmount, minSwapAmount) => {
-    return (
-      await _mintVault(vaultContract.estimateGas, swapAmount, minSwapAmount)
+    const gasEstimate = (
+      await _mintVault(vaultContract.estimateGas, swapAmount, minSwapAmount, {
+        from: account,
+      })
     ).toNumber()
+
+    return parseInt(
+      gasEstimate +
+        Math.max(
+          mintAbsoluteGasLimitBuffer,
+          gasEstimate * mintPercentGasLimitBuffer
```

Comment:
Haha interesting I almost changed the code exactly to what you propose when doing this PR. Then I remembered that when we've been testing gas problems some time ago, we've decided that for mints on the vault we increase the gasLimit by either 10% of the estimated mint gasLimit or 20k. Whichever of those two is bigger.

I must admit I haven't re-inspected the current gas spending on mints to see if it would make sense to simplify that code to just calling `increaseGasLimitByBuffer `

Reply:
oh you are right - I was confused. It is gasEstimate + a buffer...